bin/cigar_parser.R

```diff
@@ -0,0 +1,1188 @@
+        } else {
+            return(insertion_row)
+        }
+
+    } else {
+        return(insertion_row)
+    }
+}
+
+
+#######
+### Get SNPs of the reference region
+#######
+get_SNPs <- function(fasta_coords_name, genome_ref, filter=0.0001){
+    if (genome_ref == "human" || genome_ref == "mouse") {
+        chr <- sapply(strsplit(as.character(fasta_coords_name@id),"\\:"), `[`, 1)
+        chr_num <- sapply(strsplit(chr,"\\chr"), `[`, 2)
+        coords <- sapply(strsplit(as.character(fasta_coords_name@id),"\\:"), `[`, 2)
+        start <- sapply(strsplit(coords,"\\-"), `[`, 1)
+        end <- sapply(strsplit(coords,"\\-"), `[`, 2)
+
+        ### Upload the required database from biomaRt
+        if (genome_ref == "human"){
+            snp_dataset = "hsapiens_snp"
+        } else {
+            snp_dataset = "mmusculus_snp"
+        }
+        snp.db <- useMart("ENSEMBL_MART_SNP", dataset=snp_dataset)
+        snps <- getBM(c("refsnp_id", "allele", "chr_name", "chrom_start", "chrom_end", "minor_allele_freq"),
+                                    filters = c("chr_name", "start", "end"),
+                                    values = list(chr_num,as.numeric(start),as.numeric(end)), mart = snp.db)
+    } else {
+        chr <- "chr?"
+        start <- 0
+        snps <- data.frame(refsnp_id=c(), allele=c(), chr_name=c(), chrom_end=c(), minor_allele_freq=c())
+    }
+    return(snps %>% filter(minor_allele_freq >= filter))
+    # snps %>% filter(minor_allele_freq >= 0.001) #With this filter, it just appear those SNPs that USCS genome shows you
+    # snps_filt <- snps %>% filter(minor_allele_freq >= 0.0001) # 1 of each 10,000
+}
+
+#########
+### Get substitutions (% by position) using pileup
+#########
+get_substitutions_pileup <- function(bam_file){
+    #### Get bam file and generate pileup
+    p_param <- PileupParam(min_mapq=13,
+        min_base_quality=10,
+        min_nucleotide_depth=4)
+    res <- pileup(bam_file, pileupParam = p_param)
+    res <- res %>% group_by(pos, nucleotide) %>% dplyr::summarise(count = sum(count)) ### to join counts of fw with rv alignment and avoid position and nucleotides duplicates
+    ### Get percentages
+    fperc <- res %>% group_by(pos) %>% transmute(percentage = round((count / sum(count))*100, 2))
+    fperc$nucleotide <- res$nucleotide
+    ### Return data frame with all positions and the percentage of each nucleotide by position
+    return(fperc)
+}
+
+#############
+### Template-based quantification
+#############
+templateCount <- function(ori_ref_file, new_ref_temp, template_bam, readsAlnInfo_collaps, sampleID, alnCompleteInfo_corrected, alnCompleteInfo_NOTcorrected){
```

Comment:
Position was not taken into account. In test template-based edits were not reported since reads alignment start position is different to template start alignment position.

Reply:
This should be solved with the code suggestions. When the start position of the alignment was not used, the template reference alignment was not equivalent to the alignment of reads that had been edited by the template.

Reply:
Ah I see, [this](https://github.com/nf-core/crisprseq/pull/1/files#diff-bf809c928cb10b54d251dec8a140a2d5505150f97175d8d7b51dda9cc57971feR172) change should fix that issue, is that right? Thanks!